src/Fmt_ast.ml

```diff
@@ -950,7 +958,10 @@ and fmt_pattern c ?pro ?parens ({ctx= ctx0; ast= pat} as xpat) =
                   let pro =
                     if first_grp && first then pro0 $ open_hovbox (-2)
                     else if first then proI $ open_hovbox (-2)
-                    else pro2
+                    else
+                      pro2
```

Comment:
This is called once, there is no need for the optional argument. And it does not format well.